src/KeyStoreUtils.ts

```diff
@@ -134,9 +145,35 @@ export namespace KeyStoreUtils {
             messageBytes = Buffer.from(message, 'utf8');
         }
 
-        const sig = TezosMessageUtils.writeSignatureWithHint(signature, 'edsig');
-        const pk = TezosMessageUtils.writeKeyWithHint(publicKey, 'edpk');
+        return checkSignature(signature, messageBytes, publicKey);
+    }
+
+    export async function checkSignature(signature: string, bytes: Buffer, publicKey: string): Promise<boolean> {
+        const sigPrefix = signature.slice(0, 5);
+        const keyPrefix = publicKey.slice(0, 4);
+        let curve = SignerCurve.ED25519;
+
+        if (sigPrefix === 'edsig' && keyPrefix === 'edpk') {
+            curve = SignerCurve.ED25519;
+        } else if (sigPrefix === 'spsig' && keyPrefix === 'sppk') {
+            curve = SignerCurve.SECP256K1;
+        } else if (sigPrefix === 'p2sig' && keyPrefix === 'p2pk') {
+            throw new Error('secp256r1 curve is not currently supported');
+        } else {
+            throw new Error(`Signature/key prefix mismatch ${sigPrefix}/${keyPrefix}`);
+        }
+
+        const sig = TezosMessageUtils.writeSignatureWithHint(signature, sigPrefix);
+        const pk = TezosMessageUtils.writeKeyWithHint(publicKey, keyPrefix);
+
+        if (curve === SignerCurve.ED25519) {
+            return await CryptoUtils.checkSignature(sig, bytes, pk);
+        }
+
+        if (curve === SignerCurve.SECP256K1) {
+            return secp256k1.ecdsaVerify(sig, bytes, pk);
```

Comment:
maybe also wrap this in CryptoUtils

Reply:
CryptoUtils is a wrapper for libsodium mostly, which doesn't support this curve.

Reply:
oh, thought CryptoUtils was vendoring the crypto libraries

Reply:
We mostly just work with ed25519 for tz1 accounts.